app/assets/javascripts/discourse/app/lib/screen-track.js

```diff
@@ -85,6 +89,99 @@ export default class {
     this._anonCallback = cb;
   }
 
+  consolidateTimings(timings, topicTime, topicId) {
+    this._consolidatedTimings = this._consolidatedTimings || [];
+
+    let found = this._consolidatedTimings.find((elem) => elem[2] === topicId);
+
+    if (found) {
+      this._consolidatedTimings.removeObject(found);
+
+      const oldTimings = found[0];
+      Object.keys(oldTimings).forEach((id) => {
+        if (timings[id]) {
+          oldTimings[id] += timings[id];
+        }
+      });
+      found[1] += topicTime;
+      found[0] = Object.assign({}, timings, found[0]);
+    }
+
+    found = found || [timings, topicTime, topicId];
```

Comment:
I would recommend making a new class, or use an object literal instead of a tuple here. It's confusing to see all the array indices above `elem[2]`, `found[1]` and not know what attributes they represent.

Reply:
done moved to an object